spec/inputs/tcp_spec.rb

```diff
@@ -646,6 +650,75 @@ def get_port
               expect(result.first.get("message")).to eq(message)
             end
           end
+
+          context "with enforced protocol version" do
+            let(:config) do
+              base_config.merge 'ssl_supported_protocols' => [ tls_version ]
+            end
+
+            # when (Docker) testing against LS 6.x it's using Java 1.8 where TLS 1.3 isn't available
+            let(:tls_version) { (JOpenSSL::VERSION > '0.12' && TcpHelpers.tls13_available_by_default?) ? 'TLSv1.3' : 'TLSv1.2' }
+
+            it "should be able to connect and write data" do
+              used_tls_protocol = nil
+              result = TcpHelpers.pipelineless_input(subject, 1) do
+                sslsocket.connect
+                sslsocket.write("#{message}\n")
+                used_tls_protocol = sslsocket.session.to_java(javax.net.ssl.SSLSession).getProtocol
+                tcp.flush
+                sslsocket.close
+                tcp.close
+              end
+              expect(result.size).to eq(1)
+              expect(used_tls_protocol).to eql tls_version
+            end
+          end
+
+          context "with enforced protocol range" do
+            let(:config) do
+              base_config.merge 'ssl_supported_protocols' => [ 'TLSv1.3', 'TLSv1.2' ]
+            end
+            let(:sslcontext) do
+              super().tap { |ctx| ctx.ssl_version = 'TLSv1.2' }
+            end
+
+            it "should be able to connect and write data" do
+              used_tls_protocol = nil
+              result = TcpHelpers.pipelineless_input(subject, 1) do
+                sslsocket.connect
+                sslsocket.write("#{message}\n")
+                used_tls_protocol = sslsocket.session.to_java(javax.net.ssl.SSLSession).getProtocol
+                tcp.flush
+                sslsocket.close
+                tcp.close
+              end
+              expect(result.size).to eq(1)
+              expect(used_tls_protocol).to eql 'TLSv1.2'
+            end
+          end
+
+          context "with specified cipher suites" do
+            let(:config) do
+              base_config.merge 'ssl_cipher_suites' => [ cipher_suite ]
+            end
+
+            let(:cipher_suite) { 'TLS_RSA_WITH_AES_128_GCM_SHA256' }
```

Comment:
similarly here, I would like a spec for behaviour when negotiation for a common cipher suite fails to ensure that our plugin handles it gracefully without crashing.